core/shared/platform/linux-sgx/sgx_socket.c

```diff
@@ -665,9 +716,26 @@ int
 os_socket_recv_from(bh_socket_t socket, void *buf, unsigned int len, int flags,
                     bh_sockaddr_t *src_addr)
 {
-    errno = ENOSYS;
+    struct sockaddr_in addr;
+    socklen_t addr_len = sizeof(addr);
+    ssize_t ret;
 
-    return BHT_ERROR;
+    if (ocall_recvfrom(&ret, socket, buf, len, flags, &addr, &addr_len,
+                       addr_len)
+        != SGX_SUCCESS) {
+        errno = ENOSYS;
+        return -1;
+    }
+
+    if (ret < 0) {
+        return ret;
+    }
+
+    if (src_addr) {
+        sockaddr_to_bh_sockaddr((struct sockaddr *)&addr, addr_len, src_addr);
```

Comment:
Should check the return value?

Reply:
Yes. I also noticed it was not handled in the POSIX implementation. I have created an additional commit with this error handling for POSIX as well.